agent/provider/tests/rule_cli.rs

```diff
@@ -194,17 +207,70 @@ fn rule_set_with_import_cert_should_add_to_keystore_and_rulestore(rule: &str, mo
         .arg(resource_cert_dir.join("foo_ca-chain.cert.pem"))
         .arg("--mode")
         .arg(mode)
+        .arg("--whole-chain")
         .assert()
         .success();
 
     let result = list_rules_command(data_dir.path());
     let added_certs = list_certs(data_dir.path());
 
     for cert in added_certs {
-        assert_eq!(result["outbound"][rule][cert]["mode"], mode);
+        let mode_actual = result["outbound"][rule]
+            .as_object()
+            .and_then(|obj| obj.iter().find(|(id, _cert)| id.starts_with(&cert)))
+            .map(|(_id, value)| &value["mode"]);
+
+        assert!(mode_actual.is_some());
+        assert_eq!(mode_actual.unwrap(), mode);
     }
 }
 
+#[test_case("partner", "all")]
+#[test_case("partner", "none")]
+#[test_case("partner", "whitelist")]
+#[serial_test::serial]
+fn rule_set_with_import_cert_chain_should_add_whole_to_keystore_and_leaf_to_rulestore(
+    rule: &str,
+    mode: &str,
+) {
+    let (data_dir, resource_cert_dir) = prepare_test_dir_with_cert_resources();
+
+    Command::cargo_bin("ya-provider")
+        .unwrap()
+        .env("DATA_DIR", data_dir.path().to_str().unwrap())
+        .arg("rule")
+        .arg("set")
+        .arg("outbound")
+        .arg(rule)
+        .arg("import-cert")
+        .arg(resource_cert_dir.join("foo_ca-chain.cert.pem"))
+        .arg("--mode")
+        .arg(mode)
+        .assert()
+        .success();
+
+    let result = list_rules_command(data_dir.path());
+    let added_certs = list_certs(data_dir.path());
+    let added_certs: HashSet<String> = HashSet::from_iter(added_certs.into_iter());
+
+    assert_eq!(added_certs.len(), 2);
```

Comment:
I do not later iterate over all certificates, instead I access them through id, so just wanted to fail it because test is expected to check all certificates. Can delete it.

Reply:
Deleted.